catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/web/metrics/IcebergMetricsManager.java

```diff
@@ -0,0 +1,184 @@
+/*
+ *  Copyright 2023 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.catalog.lakehouse.iceberg.web.metrics;
+
+import com.datastrato.gravitino.catalog.lakehouse.iceberg.IcebergConfig;
+import com.datastrato.gravitino.catalog.lakehouse.iceberg.ops.IcebergTableOps;
+import com.datastrato.gravitino.catalog.lakehouse.iceberg.web.IcebergRestUtils;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import org.apache.iceberg.metrics.MetricsReport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IcebergMetricsManager {
+  private static final Logger LOG = LoggerFactory.getLogger(IcebergTableOps.class);
+  public static final String ICEBERG_METRICS_STORAGE = "metricsStorage";
+  public static final String ICEBERG_METRICS_STORAGE_RETAIN_DAYS = "metricsStorageRetainDays";
+  public static final String ICEBERG_METRICS_QUEUE_CAPACITY = "metricsQueueCapacity";
+
+  // Register IcebergMetricsStorage's short name to its full qualified class name in the map. So
+  // that user doesn't need to specify the full qualified class name when creating an
+  // IcebergMetricsStorage.
+  private static final ImmutableMap<String, String> ICEBERG_METRICS_STORAGES =
+      ImmutableMap.of(
+          MemoryMetricsStorage.ICEBERG_METRICS_STORAGE_MEMORY_NAME,
+          MemoryMetricsStorage.class.getCanonicalName());
+
+  private final IcebergMetricsFormatter icebergMetricsFormatter;
+  private final IcebergMetricsStorage storage;
+  private final int retainDays;
+
+  private BlockingQueue<MetricsReport> queue;
+  private Thread metricsWriterThread;
+  private volatile boolean isClosed = false;
+  private ScheduledExecutorService scheduledThreadPoolExecutor;
+
+  public IcebergMetricsManager(IcebergConfig icebergConfig) {
+    icebergMetricsFormatter = new IcebergMetricsFormatter();
+    String metricsStorage = icebergConfig.get(IcebergConfig.ICEBERG_METRICS_STORAGE);
+    storage = loadIcebergMetricsStorage(metricsStorage);
```

Comment:
Be better to rename "Storage" to "Store" here and in other places. 

Also here `storage = loadXXXStorage(xxxStorage)`, so much "storage" makes me quite confused.

Reply:
done